### PR TITLE
Date range primitive

### PR DESCRIPTION
**Pull request by JohnC-80**
# New component: `<DateRangeWrapper>`

## Purpose
A primitive wrapper component for assembling date range structures with 'start' and 'end' fields. Layout and mark-up could vary for this combination of fields - so that aspect is kept variable with this approach.

## Approach
Uses render-props to supply `exclude` props for dates outside of the range.
Has props on the main component to override/augment existing functionality to handle edge use-cases.

[Check out the readme](https://github.com/folio-org/stripes-components/blob/date-range-primitive/lib/DateRangeWrapper/readme.md)

daterangewrapper

## Tests are there
image

### CODE REVIEW

File: lib/DateRangeWrapper/DateRangeWrapper.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const callAll = (...funcs) => (...args) => funcs.forEach(fn => fn && fn(...args));
```
Comment (mkuklis) [inline lib/DateRangeWrapper/DateRangeWrapper.js]: @JohnC-80 this one is pretty cool 👍

Comment (mkuklis) [reply lib/DateRangeWrapper/DateRangeWrapper.js]: Everything looks great @JohnC-80

**Comment by cherewaty** [comment]
This looks pretty good. I'm going to try it out in eHoldings before giving it a full thumbs up.

File: lib/DateRangeWrapper/DateRangeWrapper.js (hunk 1 above)
Comment (JohnC-80) [reply lib/DateRangeWrapper/DateRangeWrapper.js]: That's a trick from Kent Dodd's. Gotta love elegant one liners. :smiley_cat:

File: lib/DateRangeWrapper/readme.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// the only child is a function, mentioned later as the 'child function'
+<DateRangeWrapper>
+  {({
+    getStartInputProps,
+    getEndInputProps,
+    endDateExclude,
+    startDateExclude,
+  }) => (
+    <div>
+      <Datepicker
```
Comment (cherewaty) [inline lib/DateRangeWrapper/readme.md]: We probably want to deprecate the `input` prop of `Datepicker` when we start to use `reduxFormField()` with it.

Is this possible instead?
```
<Field
  component={Datepicker}
  label="Start date"
  exclude={startDateExclude}
  {...getStartInputProps}
/>
```

...then later we can update the documentation to use a bare `Datepicker` without `redux-form`?

Comment (JohnC-80) [reply lib/DateRangeWrapper/readme.md]: With Redux-form/Field, that's the way to do it. I'll include an additional 'with redux-form' example.

**Pull request merged**